backend/README.md

```diff
@@ -53,3 +52,36 @@ From the `backend` directory:
 ```
 yarn deploy
 ```
+
+# Python cloud functions
+
+[Python cloud functions](https://cloud.google.com/functions/docs/quickstart-python) are not
+administered by Firebase, though they are associated
+with the same project specified above, so you will need the same access to administer them.
+
+They live in `backend/python-functions/main.py`.
+
+## Testing locally
+
+In `python-functions` set up your virtual environment:
+
+```sh
+virtualenv venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
+Unlike Firebase, we don't have tools to emulate these functions locally.
+But we can write tests against them, in `python-functions/test.py`. Run them with:
+
+```sh
+python -m unittest
+```
+
+## Deployment
+
+Deploy the functions using the `gcloud` command line tool for Google Cloud Platform.
+
+```sh
+gcloud functions deploy <FUNCTION_NAME> --runtime python37 --trigger-http --allow-unauthenticated
```

Comment:
I haven't thought deeply about this in relation to the JS function since it has not changed since initial deploy, but what is your thinking regarding deploys? We could have one staging function and one production function, then set the URL as an environment variable and have staging/master and production configure those vars individually?

Reply:
hmmmm ..... i definitely agree that this deploy process leaves something to be desired. I'm not sure how we would vary the configuration by branch (let me know if there is something specific you had in mind?) but it shouldn't be too tough to write our own little deploy scripts that would take an argument to specify the environment. so like `python deploy.py production` or whatever ... I will take a crack at that

Reply:
I am thinking we would use environment variables to vary by conceptual tier, in one of two ways:
1. Production vs Non-Production
2. Production vs Staging vs Local

I would go with option 1 for now because I don't see enough value in option 2 to outweigh the additional complexity. So then we'd use an env var that points to the right Cloud Function path and set that env var directly in Now.sh production to point to the prod path, and set it elsewhere to point to the stage path. The only thing to figure out there would be how to ensure that all Now.sh deployments _except_ production share the same env var. But that should be an imminently solvable problem!

Reply:
ah yes got it. that is actually pretty much what I was thinking (and have been working on), I think I wasn't sure whether you were talking about the envvar would also affect the function deploy process itself!